### ISSUE TITLE
[Coscheduling] Issues about priority preemption

### ISSUE THREAD

**Issue by zhiyxu** [kubernetes-sigs/scheduler-plugins#443]
Recently, we encountered some issues in the use of the Coscheduling plugin, most of which are related to priority and preemption, I want to give some feedbacks for those using this plugin, these issues are as follows:

- Issue 1: Distributed tasks are constantly scheduled and this totally block scheduling of other tasks. #429  (backoffQueue can solve this problem)
- Issue 2: Priority preemption is currently not supported. #428 

However, priority preemption is necessary for the scheduling of distributed tasks, but after turn on priority preemption(in our environment), the following problems will arise:

- Issue 3: Even if the cluster resources are sufficient in total, but the resources suitable for distributed tasks' nodeAffinity are insufficient, so distributed tasks can't start, but they would still enter PostFilter, and the low priority tasks will be preempted repeatedly.
- Issue 4: When low-priority tasks are preempted, they may need a long period to terminate, distributed tasks may timeout in Permit extension point and move to backoff queue, then newly created low-priority tasks would occupied the resource and running again, in next scheduling cycle of distributed task, these newly created low-priority tasks would be preempted again, and distributed tasks still can't run. #431  (could solved by extend timeout in Permit)

The main reason for the above issues is that the resource check in PreFilter cannot fully guarantee that distributed tasks can run as a whole. Even if the total resources of the entire cluster meet minResource, distributed tasks may still can't run due to resource fragmentation, nodeAffinity and other reasons. Once the distributed tasks which can't run as a whole pass PreFilter, it's the start of the issue, it will eventually lead to scheduling blocking and unnecessarily preemption of low-priority tasks.

One possible solution is that, PreFilter extension point need to check whether the distributed tasks can finally run, only if the tasks can run, they can continue the scheduling cycle, otherwise it will terminate as a whole, i.e achieve all or nothing in PreFilter. But under current scheduler framework, it's still hard to check everything in PreFilter, maybe we need to totally solve these problems in framework level.

In addition, in our scenario, CapacityScheduling and Coscheduling need to be used together, the mixup of plugins also brings a lot of trouble, because the preemption rules based on priority and min-max quota are too complicated.

To sum up, when resources are sufficient and priority preemption is not required, Coscheduling works fine; but when it comes to priorities, Coscheduling performs not well.

**Comment by Huang-Wei** [comment]
/kind feature

> To sum up, when resources are sufficient and priority preemption is not required, Coscheduling works fine; but when it comes to priorities, Coscheduling performs not well.

Thanks for the detailed testing and reporting.
